extensions/msal-node-extensions/src/dpapi-addon/dpapi_win.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+ // Implementation referenced from https://github.com/bradhugh/node-dpapi
+
+#include <node.h>
+#include <nan.h>
+#include <Windows.h>
+#include <dpapi.h>
+#include <functional>
+#include <iostream>
+#include <string>
+
+v8::Local<v8::String> CreateUtf8String(v8::Isolate* isolate, char* strData)
+{
+	return v8::String::NewFromUtf8(isolate, strData, v8::NewStringType::kNormal).ToLocalChecked();
+}
+
+void ProtectDataCommon(bool protect, Nan::NAN_METHOD_ARGS_TYPE info)
+{
+	v8::Isolate* isolate = info.GetIsolate();
+
+	if (info.Length() != 3) {
+		isolate->ThrowException(v8::Exception::RangeError(
+			CreateUtf8String(isolate, "3 arguments are required")));
+	}
+
+	if (info[0]->IsNullOrUndefined() || !info[0]->IsUint8Array())
+	{
+		isolate->ThrowException(v8::Exception::TypeError(
+			CreateUtf8String(isolate, "First argument, data, must be a valid Uint8Array")));
+	}
+
+	if (!info[1]->IsNull() && !info[1]->IsUint8Array())
+	{
+		isolate->ThrowException(v8::Exception::TypeError(
+			CreateUtf8String(isolate, "Second argument, optionalEntropy, must be null or an ArrayBuffer")));
+	}
```

Comment:
I'd recommend constructing the correct C++ types ASAP after verifying the input types, that way you can work directly in C++, without having to think about v8 variadic types.